docs/reference.md

```diff
@@ -0,0 +1,240 @@
+
+```nix
+{ pkgs, ... }:
+
+let
+  doom-emacs = pkgs.callPackage (builtins.fetchTarball {
+    url = https://github.com/nix-community/nix-doom-emacs/archive/master.tar.gz;
+  }) {
+    doomPrivateDir = ./doom.d;  # Directory containing your config.el, init.el
+                                # and packages.el files
+  };
+in {
+  home.packages = [ doom-emacs ];
+}
+```
+
+
+## NixOS
+
+`File: flake.nix`
+```nix
+{
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    nix-doom-emacs.url = "github:nix-community/nix-doom-emacs";
+  };
+  
+  outputs = {
+    self,
+    nixpkgs,
+    nix-doom-emacs,
+    ...
+  }: {
+    nixosConfigurations.exampleHost = nixpkgs.lib.nixosSystem {
+      system  = "x86_64-linux";
+      modules = [
+        { 
+          environment.systemPackages = 
+            let
+              doom-emacs = inputs.nix-doom-emacs.packages.${system}.default.override {
+                doomPrivateDir = ./doom.d;
+              };
+            in [
+              doom-emacs
+            ];
+        }
+        # ...
+      ];
+    };
+  };
+}
+```
+
+You can see all overridable parameters of nix-doom-emacs in [default.nix](../default.nix).
+
+## Standalone
+
+### Flake
+
+```nix
+{
+  description = "nix-doom-emacs shell";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    nix-doom-emacs.url = "github:nix-community/nix-doom-emacs";
+  };
+
+  outputs = { self, nixpkgs, nix-doom-emacs, ... }:
+  let
+    system = "x86_64-linux";
+    pkgs = import nixpkgs { inherit system; };
+    doom-emacs = nix-doom-emacs.packages.${system}.default.override {
+      doomPrivateDir = ./doom.d;
+    };
+  in
+  {
+    devShells.${system}.default = pkgs.mkShell {
+      buildInputs = [ doom-emacs ];
+    };
+  };
+}
+```
+
+### Non-Flake
+```nix
+{ pkgs ? import <nixpkgs> { } }:
+
+let
+  repo = pkgs.fetchFromGitHub {
+    owner = "nix-community";
+    repo = "nix-doom-emacs";
+    rev = "<commit>";
+    sha256 = "<hash>";
+  };
+  nix-doom-emacs = pkgs.callPackage (import repo) {
+    doomPrivateDir = ./doom.d;
+  };
+in
+pkgs.mkShell {
+  buildInputs = [ nix-doom-emacs ];
+}
+```
+
+# Setup
+
+## Emacs daemon
+
+If you use the Home-Manager module, you can enable it via `services.emacs.enable = true;`. The Home-Manager module will do the rest for you.
+
+If you're not, and you're using a standalone method (NixOS/nix-darwin without Home-Manager) instead, you'll need:
+
+```nix
+services.emacs = {
+  enable = true;
+  package = inputs.doom-emacs.packages.${system}.doom-emacs.override {
+    doomPrivateDir = ./doom.d;
+  };
+};
+```
+
+You can now run `emacsclient -c` to connect to the daemon.
+
+## Custom Emacs derivations (i.e., PGTK, NativeComp)
+
+If you're using the Home-Manager module, you can use the `emacsPackage` attribute after applying `emacs-overlay` to your nixpkgs:
+
+```nix
+programs.doom-emacs = {
+  enable = true;
+  doomPrivateDir = ./doom.d;
+  emacsPackage = pkgs.emacsPgtkNativeComp;
```

Comment:
```
error: anonymous function at /nix/store/1q10ylzzjmxwz5a4zy3dbdd9dzw8n0l6-source/default.nix:1:1 called with unexpected argument 'emacsPackage'

       at /nix/store/4idp9fvcbrp511x4vykas35yiiaqfbw6-nixos/nixos/lib/customisation.nix:72:16:

           71|     let
           72|       result = f origArgs;
             |                ^
           73|
       Did you mean emacsPackages?
(use '--show-trace' to show detailed location information)
```

Reply:
With some more digging, it's not obvious how to compose emacs-overlay with the non-flake home manager, since one uses `home.packages`, and the other uses the module enable method.